modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobContainer.java

```diff
@@ -69,9 +71,12 @@ private InputStream openInputStream(OperationPurpose purpose, String blobName, l
         } catch (Exception e) {
             Throwable rootCause = Throwables.getRootCause(e);
             if (rootCause instanceof BlobStorageException blobStorageException) {
-                if (blobStorageException.getStatusCode() == 404) {
+                if (blobStorageException.getStatusCode() == RestStatus.NOT_FOUND.getStatus()) {
                     throw new NoSuchFileException("Blob [" + blobKey + "] not found");
                 }
+                if (length != null && blobStorageException.getStatusCode() == RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus()) {
```

Comment:
Is there a reason we're rethrowing as a `RequestedRangeNotSatisfiedException` only if length is not null?

According to the Azure document, it is possible to only set the startByte (ie, offset) without a length and if that one is invalid, the service returns a 416. That's also how the AzureHttpHandler seems to be implemented.

I think we should just rethrow a `RequestedRangeNotSatisfiedException` here, possibly passing Long.MAX_VALUE to the RequestedRangeNotSatisfiedException.

Reply:
not null check is needed to avoid NPE during constructing RequestedRangeNotSatisfiedException (it takes primitive `long` argument), also not null `length` is somewhat meaningful in the context of RequestedRangeNotSatisfiedException since this exception is thrown only if length param is of the range

I think if we pass Long.MAX_VALUE we loose piece of real info which lead to exception.

Reply:
> I think if we pass Long.MAX_VALUE we loose piece of context (real value) which led to exception.

Sorry, I was suggesting to rethrow `RequestedRangeNotSatisfiedException` also when `length` is null. 

In the code today we allow passing a `null` length down the `AzureBlobStore.getInputStream()` method which can throw a `RequestedRangeNotSatisfiedException`, but if length is null we don't rethrow it. I think we should, and to do that we need to build  a `RequestedRangeNotSatisfiedException` that requires a `long` length value: if it's null we can pass Long.MAX_VALUE or `-1`.